src/webots/nodes/utils/WbWorld.cpp

```diff
@@ -445,6 +445,8 @@ void WbWorld::createX3DMetaFile(const QString &filename) const {
             WbVector3 v = m.translation();
             if (!v.almostEquals(WbVector3()))
               deviceObject.insert("transformOffset", v.toString(WbPrecision::FLOAT_MAX));
+            if (motor)
```

Comment:
It doesn't seem correct to assume that if a device has a motor but it is not a joint, then it is a `Track` device.
A `Propeller` has a motor too, so with the current code the `Propeller` motor is tagged with the `track` flag.
I think that here an additional type check is needed to set the track flag correctly just for Track motors.

Reply:
Very good point, I did forget the propeller case.